test-ts/TrueRewardBackedToken.test.ts

```diff
@@ -643,5 +643,115 @@ describe('TrueRewardBackedToken', () => {
         })
       })
     })
+
+    /**
+     * Note: See magic numbers explanation in AssuredFinancialOpportunity tests
+     */
+    describe('award amount', () => {
+      it('0 when reward basis is 100%', async () => {
+        await token.connect(holder).enableTrueReward()
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        expect(await financialOpportunity.awardAmount()).to.eq(0)
+      })
+
+      it('properly calculated when reward basis is 70%', async () => {
+        await token.connect(holder).enableTrueReward()
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        expect(await financialOpportunity.awardAmount()).to.equal('17179876005666582700')
+      })
+    })
+
+    describe('award pool', () => {
+      beforeEach(async () => {
+        await token.connect(holder).enableTrueReward()
+      })
+
+      it('awards 0 when reward basis is 100%', async () => {
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+        await financialOpportunity.awardPool()
+
+        expect(await token.balanceOf(mockPoolAddress)).to.equal(0)
+        expect(await aaveFinancialOpportunity.getBalance()).to.equal(parseEther('150'))
+      })
+
+      it('awards proper amount', async () => {
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        expect(await token.balanceOf(mockPoolAddress)).to.equal(parseEther('0'))
+        expect(await financialOpportunity.getBalance()).to.equal('132820123994333417300')
+        expect((await financialOpportunity.getBalance()).div(await financialOpportunity.perTokenValue())).to.equal('100')
+
+        await expect(financialOpportunity.awardPool()).to.emit(financialOpportunity, 'awardPoolSuccess').withArgs('11453250670444388466')
+
+        expect(await token.balanceOf(mockPoolAddress)).to.equal('17179876005666582699')
+        expect(await financialOpportunity.getBalance()).to.equal('132820123994333417300')
+      })
+
+      it('awards 0 on subsequent calls', async () => {
+        await financialOpportunity.setRewardBasis(0.7 * 1000)
+        await lendingPoolCore.setReserveNormalizedIncome(parseEther('1500000000'))
+
+        await financialOpportunity.awardPool()
+
+        expect(await financialOpportunity.awardAmount()).to.equal(0)
+        await financialOpportunity.awardPool()
```

Comment:
```suggestion
        await expect(financialOpportunity.awardPool()).to.emit(financialOpportunity, 'awardPoolSuccess').withArgs('0')
```

Reply:
By the way, this event is triggered upon successful call to `withdraw`. I'm pretty sure not all financial opportunities will accept a withdrawal of value `0`.